doc/designs/black-codestyle.rst

```diff
@@ -0,0 +1,308 @@
+  Once the formatting changes have landed in git, IPA can provide a
+  ``.gitignorerevs`` file that lists the formatting commits from all branches.
+* |Black| sometimes creates less readable code. It's possible to disable
+  reformatting with ``# fmt: off`` / ``# fmt: on``. This document proposes a
+  short list of :ref:`exceptions <black-disable-formatting>`.
+* Reformatting does never change the meaning of code. |Black| does not
+  perform modifications that affect the AST (abstract syntax tree). By
+  default the tool compares the AST before and after reformatting to ensure
+  correctness. This is also the reason why black cannot perform string
+  concatenation and sometimes creates line like
+  ``text = "multiline " "string"``.
+* In general |Black| spreads out long expression over multiple lines and
+  therefore can increases lines of code. The slightly larger line length
+  reduces lines of code in other cases. Reformatting increases the total
+  lines of code of all ``.py`` files by about 1% (765k before, 773k after).
+
+  When I first used |Black| in personal projects I was slightly annoyed by
+  this formatting style. Shortly later I started to appreciate the style.
+  It makes it easier to recognize begin and end of a list block, reduces
+  diffs when adding/removing an item, and encourages use of keyword arguments
+  in function calls.
+
+Open questions
+==============
+
+ipa-4-6 branch
+--------------
+
+The |FreeIPA| team still maintains ipa-4-6 branch for RHEL 7. The branch
+simplifies backports and fixes of IPA in RHEL.
+
+According to the `RHEL Life Cycle`_ document, RHEL 7 is slowly winding down.
+RHEL 7.8 was released March 31, 2020. RHEL 7.9 will be the last RHEL 7
+y-stream release. RHEL 7 has reached end of full support in August 2019 and
+is going to reach end of *Maintenance Support 1* in August 2020. During the
+*Maintenance Support 2* phase critical and import security fixes and select
+urgent bug fixes are provided until June 2024.
+
+I expect that the amount of backports to 4.6 will soon diminish to a few
+commits a year.
+
+Q: When can we start to limit backports to urgent CVEs only?
+
+|Black| target version
+----------------------
+
+|Black| creates slightly different output depending on the target version
+option. With Python 2.7 as minimum target version, |Black| retains Python 2
+features like ``u""`` string prefix. With target version Python 3.6, |Black|
+removes the redundant ``u""`` and can make use of 3.6 features.
+
+Q: Should we use a ``target-version`` 3.6 to automate the next step in
+|FreeIPA|'s Python 3 migration?
+
+New code style
+==============
+
+The new code style is *whatever black does*.
+
+The rules for i18n strings and unused variables from the |FreeIPA|
+`Python code style`_ still apply, though.
+
+|Black| increases the permitted maximum line length from 79/80 characters to
+88 characters. According to `black project`_'s documentation
```

Comment:
In short: whether we approve this change or not, I would appreciate moving to 88 characters (or more (*)).

(*) Using longer lines would make code accessibility lower: longer lines are only readable on high-resolution screens and/or when using small characters. So a middle-ground has to be found, and 88c is at least better than 80c.